ckan/tests/helpers.py

```diff
@@ -316,21 +371,41 @@ def test_ckan_site_title(self):
 
     :param value: the new config key's value, e.g. ``'My Test CKAN'``
     :type value: string
+
+    .. seealso:: The context manager :py:func:`changed_config`
     '''
     def decorator(func):
+        @functools.wraps(func)
         def wrapper(*args, **kwargs):
-            _original_config = config.copy()
-            config[key] = value
+            with changed_config(key, value):
+                return func(*args, **kwargs)
+        return wrapper
+    return decorator
 
-            try:
-                return_value = func(*args, **kwargs)
-            finally:
-                config.clear()
-                config.update(_original_config)
 
-            return return_value
-        return nose.tools.make_decorator(func)(wrapper)
-    return decorator
+@contextlib.contextmanager
+def changed_config(key, value):
```

Comment:
This looks a lot like https://github.com/ckan/ckan/blob/a77ff9c021c80eee949d2ed11cbafd743ee5c04a/ckan/tests/helpers.py#L301-L333